File: core/src/jvmMain/java/com/xebia/functional/xef/auto/jvm/Description.java
package com.xebia.functional.xef.auto.jvm;

import java.lang.annotation.*;

@Target({ElementType.TYPE, ElementType.ANNOTATION_TYPE, ElementType.FIELD, ElementType.METHOD, ElementType.PARAMETER})
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface Description {
    String value();
}

File: core/src/commonMain/kotlin/com/xebia/functional/xef/auto/Description.kt
package com.xebia.functional.xef.auto

import kotlinx.serialization.ExperimentalSerializationApi
import kotlinx.serialization.SerialInfo

/**
 * Description of a property to enhance the LLM prompt chances to get the data in the right format
 */
@OptIn(ExperimentalSerializationApi::class)
@SerialInfo
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.PROPERTY, AnnotationTarget.FIELD)
expect annotation class Description(val value: String)
